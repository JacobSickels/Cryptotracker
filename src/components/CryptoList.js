import React from 'react';
import { connect } from 'react-redux';
import selectCryptos from '../selectors/cryptos';

import CryptoListItem from './CryptoListItem';

export const CryptoList = (props) => (
    <div> 
        <h2>CryptoList</h2> 
        Bitcoin:
        <CryptoListItem crypto={props.cryptos.bitcoin} />
        Litecoin:
        <CryptoListItem crypto={props.cryptos.litecoin} /> 
        Ethereum:
        <CryptoListItem crypto={props.cryptos.ethereum} />
    </div>
);

const mapStateToProps = (state) => {
    return {
        cryptos: selectCryptos(state.cryptos, state.filters)
    };
};

export default connect(mapStateToProps)(CryptoList);