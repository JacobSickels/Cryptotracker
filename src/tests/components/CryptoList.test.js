import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';

import CryptoList from '../../components/CryptoList';

test('Should render CryptoList correctly', () => {
    const wrapper = shallow(<CryptoList />);
    expect(wrapper).toMatchSnapshot();
});