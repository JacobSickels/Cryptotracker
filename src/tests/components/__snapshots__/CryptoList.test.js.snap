// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CryptoList correctly 1`] = `
<div>
  <Connect(CryptoItem)
    name="Bitcoin"
  />
  <Connect(CryptoItem)
    name="Litecoin"
  />
  <Connect(CryptoItem)
    name="Ethereum"
  />
</div>
`;
